### PR TITLE
[msbuild] Add a property that specifies the final path to the Info.plist in the app bundle

### PR DESCRIPTION
**Pull request by rolfbjarne**
* Add an '_AppBundleManifest' property that specifies the final path to the
  Info.plist in the app bundle.
* Remove the '_AppBundleManifestPath' property, it's not used anywhere.
* Adjust the CompileAppManifest task to take the final path to the Info.plist,
  instead of computing it and returning it. This way the CompileAppManifest
  task does not output anything back into MSBuild (which is important, because
  the CompileAppManifest task won't execute if the output file is up-to-date,
  and if it's not executed, any output properties won't be set either).

### CODE REVIEW

File: msbuild/Xamarin.MacDev.Tasks.Core/Tasks/CompileAppManifestTaskBase.cs [hunk 1]
```diff
@@ -73,13 +73,6 @@ public abstract class CompileAppManifestTaskBase : XamarinTask
 		public string TargetArchitectures { get; set; }
 		#endregion
 
-		#region Outputs
-
-		[Output]
-		public ITaskItem CompiledAppManifest { get; set; }
```
Comment (emaf) [inline msbuild/Xamarin.MacDev.Tasks.Core/Tasks/CompileAppManifestTaskBase.cs:79]: This output property was creating an empty Info.plist file on Windows for the Inputs/Outpus MSBuild check. If we don't want to have output props on this task we can run the `Touch` task after `CompileAppManifest`.

In my opinion having an ITaskItem output prop is still the best option, that way we ensure every time this task is run the file will be created on Windows.

**Review by emaf — CHANGES_REQUESTED** [review]

File: msbuild/Xamarin.MacDev.Tasks.Core/Tasks/CompileAppManifestTaskBase.cs (hunk 1 above)
Comment (rolfbjarne) [reply msbuild/Xamarin.MacDev.Tasks.Core/Tasks/CompileAppManifestTaskBase.cs:79]: @emaf I made it an [Output] property:

https://github.com/xamarin/xamarin-macios/blob/d02e405963d3ec367fba2a583a04537648dd7b1d/msbuild/Xamarin.MacDev.Tasks.Core/Tasks/CompileAppManifestTaskBase.cs#L37-L39

would that work?

Comment (emaf) [reply msbuild/Xamarin.MacDev.Tasks.Core/Tasks/CompileAppManifestTaskBase.cs:79]: Yes, that should work, but the type should be `ITaskItem` :)

**Review by dalexsoto — APPROVED** [review]

File: msbuild/Xamarin.MacDev.Tasks.Core/Tasks/CompileAppManifestTaskBase.cs (hunk 1 above)
Comment (rolfbjarne) [reply msbuild/Xamarin.MacDev.Tasks.Core/Tasks/CompileAppManifestTaskBase.cs:79]: @emaf ok, changed!

**Review by spouliot — APPROVED** [review]

**Comment by rolfbjarne** [comment]
Test failures are unrelated:

* framework-test/Mac Catalyst/Debug: https://github.com/xamarin/maccore/issues/2414
* monotouch-test/tvOS - simulator/Debug [dotnet]: https://github.com/xamarin/maccore/issues/2304

**Review by emaf — APPROVED** [review]

**Pull request merged**